modyn/trainer_server/internal/trainer/maybe_measure_gpu_ops.py

```diff
@@ -0,0 +1,31 @@
+from types import TracebackType
+from typing import Optional, Type
+
+import torch
+from modyn.common.benchmark import Stopwatch
+
+
+class MaybeMeasureGPUOps:
+    def __init__(  # type: ignore[no-untyped-def]
+        self, measure: bool, measurement_name: str, device: str, stop_watch: Stopwatch, **kwargs
+    ) -> None:
+        self._device = device
+        self._measure = measure
+        self._measurement_name = measurement_name
+        self._kwargs = kwargs
+        self._stop_watch = stop_watch
+
+    def __enter__(self) -> None:
+        if self._measure:
+            torch.cuda.synchronize(device=self._device)
+            self._stop_watch.start(name=self._measurement_name, **self._kwargs)
```

Comment:
Can we always measure the stopwatch, and have only cuda synchronize wrapped in the if statement? I understand the numbers might not make full sense but in the past I still sometimes took a look to get an indication whether _something_ is happening. Thanks!

Reply:
done